lib/Utility/AntiMalware/Providers/Sophos.php

```diff
@@ -0,0 +1,213 @@
+ *
+ * You can contact SugarCRM, Inc. headquarters at 10050 North Wolfe Road,
+ * SW2-130, Cupertino, CA 95014, USA. or at email address contact@sugarcrm.com.
+ *
+ * The interactive user interfaces in modified source and object code versions
+ * of this program must display Appropriate Legal Notices, as required under
+ * Section 5 of the GNU Affero General Public License version 3.
+ *
+ * In accordance with Section 7(b) of the GNU Affero General Public License version 3,
+ * these Appropriate Legal Notices must retain the display of the "Powered by
+ * SugarCRM" logo and "Supercharged by SuiteCRM" logo. If the display of the logos is not
+ * reasonably feasible for technical reasons, the Appropriate Legal Notices must
+ * display the words "Powered by SugarCRM" and "Supercharged by SuiteCRM".
+ */
+
+namespace SuiteCRM\Utility\AntiMalware\Providers;
+
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerInterface;
+use SuiteCRM\Exception\Exception;
+use SuiteCRM\Exception\InvalidArgumentException;
+use SuiteCRM\Exception\MalwareFoundException;
+use SuiteCRM\Interfaces\AntiMalwareFileScanner;
+use SuiteCRM\Utility\AntiMalware\AntiMalwareTrait;
+use SuiteCRM\Utility\Configuration;
+use SuiteCRM\Utility\SuiteLogger;
+
+/**
+ * Class Sophos
+ * @package SuiteCRM\Utility\AntiMalware\Providers
+ */
+class Sophos implements AntiMalwareFileScanner, LoggerAwareInterface
+{
+    use AntiMalwareTrait;
+
+    /**
+     * @var bool $enabled
+     */
+    private $enabled = false;
+
+    /**
+     * @var string $path
+     */
+    private $path = '/opt/sophos-av/bin/savscan';
+
+    /**
+     * @var string $options
+     */
+    private $options = '-ss';
+
+    /**
+     * @var  LoggerInterface
+     */
+    private $logger;
+
+    public function __construct()
+    {
+        $globalConfig = new Configuration();
+        $configuration =
+            $globalConfig['anti_malware_scanners'][self::class] ?? null;
+
+        if (!empty($configuration)) {
+            $this->enabled = $configuration['enabled'] ?? $this->enabled;
+            $this->path = $configuration['path'] ?? $this->path;
+            $this->options = $configuration['options'] ?? $this->options;
+        }
+
+        $this->logger = new SuiteLogger();
+    }
+
+    /**
+     * Is the Anti-Malware FileScanner enabled and configured
+     * @return bool true === yes, false === no
+     */
+    public function isAntiMalwareScannerAvailable(): bool
+    {
+        if ($this->isEnabled()) {
+            return file_exists($this->getExecutablePath());
+        }
+
+        return false;
+    }
+
+    /**
+     * @param string $path - location in the file system
+     * @throws InvalidArgumentException
+     * @throws MalwareFoundException|Exception
+     */
+    public function scanFilePath(string $path): void
+    {
+        $this->validatePath($path);
+
+        $command = $this->getExecutablePath() . ' ' . $this->getOptions() . ' ' . $path;
+        $lineOutput = shell_exec($command);
```

Comment:
Almost missed this one 😅, If I'm understanding how the command is built here then I think we could run into security issues here.

E.g. What if I named my file something like ```&& rm -rf .*```.